src/Monitor.c

```diff
@@ -48,8 +53,72 @@ static bool handle_activation_log_cyclic_buffer(const enum interfaces_enum inter
     return true;
 }
 
+static bool cpu_usage_visitor(Thread_Control *the_thread, void *arg)
+{
+    float usage_percent;
+	uint32_t integer_val;
+	uint32_t float_val;
+    Timestamp_Control uptime;
+	const Timestamp_Control used_time =
+	    _Thread_Get_CPU_time_used_after_last_reset(the_thread);
+
+	_TOD_Get_uptime(&uptime);
+	_Timestamp_Subtract(&uptime_at_last_reset, &uptime, &total_usage_time);
+	_Timestamp_Divide(&used_time, &total_usage_time, &integer_val, &float_val);
+
+	usage_percent = (float)float_val / TOD_NANOSECONDS_PER_MICROSECOND;
+	usage_percent += (float)integer_val;
+
+	if (usage_percent < idle_cpu_usage_data.minimum_cpu_usage) {
+		idle_cpu_usage_data.minimum_cpu_usage = usage_percent;
+	}
+
+	if (usage_percent > idle_cpu_usage_data.maximum_cpu_usage) {
+		idle_cpu_usage_data.maximum_cpu_usage = usage_percent;
+	}
+
+	idle_cpu_usage_data.average_cpu_usage =
+	    idle_cpu_usage_data.average_cpu_usage +
+	    (usage_percent - idle_cpu_usage_data.average_cpu_usage) /
+		(benchmarking_ticks + 1);
+
+    // only first idle thread is needed, stop iteration after first step
+    return true;
```

Comment:
Are you sure about that? What about querying interfaces?

Reply:
Only data for idle thread are calculated, there is no need for calculations for other threads.